components/tidb_query/src/rpn_expr/impl_time.rs

```diff
@@ -128,13 +128,37 @@ pub fn year(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Option<Int>>
     Ok(Some(Int::from(t.year())))
 }
 
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn month_name(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Option<Bytes>> {
+    match t {
+        Some(t) => {
+            let month = t.month() as usize;
+            if t.is_zero() && ctx.cfg.sql_mode.contains(SqlMode::NO_ZERO_DATE) {
```

Comment:
`if t.invalid_zero()` is enough.

Reply:
I tried it, only on zero date and turned on NO_ZERO_DATE SqlMode. mysql will report warning.

mysql> show variables like "sql_mode";
+---------------+----------------------------------+
| Variable_name | Value                            |
+---------------+----------------------------------+
| sql_mode      | STRICT_TRANS_TABLES,NO_ZERO_DATE |
+---------------+----------------------------------+
1 row in set (0.01 sec)

mysql> select monthname('2019-00-00');
+-------------------------+
| monthname('2019-00-00') |
+-------------------------+
| NULL                    |
+-------------------------+
1 row in set (0.00 sec) 

mysql> select monthname('0000-00-00');
+-------------------------+
| monthname('0000-00-00') |
+-------------------------+
| NULL                    |
+-------------------------+
1 row in set, 1 warning (0.00 sec)

Reply:
Here I prefer remove `ctx.cfg.sql_mode.contains(SqlMode::NO_ZERO_DATE)`.